days/day-08/solutions/day08.sml

```diff
@@ -0,0 +1,65 @@
+use "../../../util.sml";
+
+local
+    structure IS = IntSet
+in
+
+datatype Op = ACC | JMP | NOP
+
+fun opFromString "acc" = ACC
+  | opFromString "jmp" = JMP
+  | opFromString "nop" = NOP
+  | opFromString s = raise Fail $ "opFromString - bad input: " ^ s
```

Comment:
I learned today that this construct is called a `discriminated union`-type (the thing with the `|`)